consensus/src/chained_bft/block_storage/block_tree.rs

```diff
@@ -158,26 +162,40 @@ where
         }
     }
 
+    // This method will only be used in this module.
+    fn get_linkable_block(&self, block_id: &HashValue) -> &LinkableBlock<T> {
+        self.id_to_block
+            .get(block_id)
+            .expect("linkable block doesn't exist.")
+    }
+
+    // This method will only be used in this module.
+    fn try_get_linkable_block_mut(
+        &mut self,
+        block_id: &HashValue,
+    ) -> Option<&mut LinkableBlock<T>> {
+        self.id_to_block.get_mut(block_id)
+    }
+
+    // This method will only be used in this module.
+    fn linkable_root(&self) -> &LinkableBlock<T> {
+        self.get_linkable_block(&self.root_id)
+    }
+
     fn remove_block(&mut self, block_id: HashValue) {
         // Remove the block from the store
         self.id_to_block.remove(&block_id);
         self.id_to_votes.remove(&block_id);
         self.id_to_quorum_cert.remove(&block_id);
     }
 
-    pub(super) fn block_exists(&self, block_id: &HashValue) -> bool {
-        self.id_to_block.contains_key(block_id)
-    }
-
     fn get_block(&self, block_id: &HashValue) -> Arc<ExecutedBlock<T>> {
         self.try_get_block(block_id)
             .expect("Block doesn't exist. Use try_get_block if None is a possible return value")
     }
 
-    fn get_linkable_block(&self, block_id: &HashValue) -> &LinkableBlock<T> {
-        self.id_to_block
-            .get(block_id)
-            .expect("linkable block doesn't exist.")
+    pub(super) fn block_exists(&self, block_id: &HashValue) -> bool {
+        self.id_to_block.contains_key(block_id)
     }
 
     pub(super) fn try_get_block(&self, block_id: &HashValue) -> Option<Arc<ExecutedBlock<T>>> {
```

Comment:
rename the function to sth like `try_get_executed_block`? also it probably should delegate to `try_get_linkable_block`